src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -172,7 +174,11 @@ public function searchBarAction(Request $request): Response
 
         $this->cspHandler?->disableCsp();
 
-        $session = $request->hasSession() ? $request->getSession() : null;
+
+        $session = null;
+        if ($request->attributes->getBoolean('_stateless') && $request->hasSession()) {
```

Comment:
I know I am late, but I thought posting it here was less interruptive then creating a new issue or discussion for this question: 

The PR touches 3 placed where it gets the session and adds checks for the `_stateless` request attribute. 
Two checks check for `!stateless && hasSession` while one of them (in `searchBarAction`)  checks for `stateless && hasSession` (without the `!`). 

I found this while debugging for what starts `Session was used while the request was declared stateless.` when calling the profiler. As far as I can tell, the only place using the session there is the this spot. 

This line looks incorret to me, I think its missing the `!` in front.

Reply:
Just tested and you’re not wrong. Maybe you really are the first to hit this case :sweat_smile:

Reply:
If I understand the implications of this correct, then that means that for users that have `stateless: false`, restoring the search bar state from session should not work, as it would not use the session in those cases. It would only use the session if `stateless: true`  (e.g. when there should be no session). Correct?

Reply:
> for users that have `stateless: false`, restoring the search bar state from session should not work

Yes, that also includes users who did not set `_stateless`. Are you up for a PR? Looks like you already know the fix!